packages/grid/_modules_/grid/hooks/features/infiniteLoader/useGridInfiniteLoader.ts

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { optionsSelector } from '../../utils/optionsSelector';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { useGridSelector } from '../core/useGridSelector';
+import { gridContainerSizesSelector } from '../../../components/GridViewport';
+import { useNativeEventListener } from '../../root/useNativeEventListener';
+import { GRID_SCROLL } from '../../../constants/eventsConstants';
+
+export const useGridInfiniteLoader = (
+  windowRef: React.MutableRefObject<HTMLDivElement | null>,
+  apiRef: GridApiRef,
+): void => {
+  const options = useGridSelector(apiRef, optionsSelector);
+  const containerSizes = useGridSelector(apiRef, gridContainerSizesSelector);
+  const isInScrollBottomArea = React.useRef<boolean>(false);
+
+  const handleGridScroll = React.useCallback(
+    (event) => {
+      const scrollPositionBottom =
+        event.target.scrollTop + containerSizes?.windowSizes.height + options.scrollBottomThreshold;
+
+      if (containerSizes && scrollPositionBottom >= containerSizes.dataContainerSizes.height) {
+        if (!isInScrollBottomArea.current && options.onScrollBottom) {
+          isInScrollBottomArea.current = true;
+          options.onScrollBottom();
+        }
+      } else {
+        isInScrollBottomArea.current = false;
+      }
+    },
+    [options, containerSizes, isInScrollBottomArea],
+  );
+
+  useNativeEventListener(apiRef, windowRef, GRID_SCROLL, handleGridScroll, { passive: true });
```

Comment:
there is already a scroll event in the grid, why not use it?
```
  useGridApiEventHandler(api as GridApiRef, GRID_SCROLLING, handleScrolling);
```

This is also a `state.isScrolling` that let you know when the grid is scrolling. We could add the position...

Reply:
I don't get is, why should we care about the scrolling state?

Reply:
Ok, I can explain. So, yes @dtassone there is a scrolling event on the grid that returns the top and left positions of the scroll. The problem is that when the virtualization kicks in those values are being reset back to 0 so it isn't possible to know the actual scroll top for the entire grid viewport. That's why I'm using this one. it works the same way and it is not fired any more times than the other but it returns the correct position.

Regarding the `isScrolling` I also don't see a benefit of adding it.

Reply:
I see the issue, that's because it's the position of the rendering zone. Maybe we should modify it to add the window scroll position so you could hook into this one for the case we are covering in this PR and then we already have the base for the case of infinite scroll server side hooked into virtualisation...

> Regarding the isScrolling I also don't see a benefit of adding it.

It's just an approach that use the state instead of the event... Probably better perf with event on this one.

Reply:
the main reason for not attaching the event to window is to decouple the rendering and from the feature.
So if I refactor the rendering engine and remove the window then I will have to refactor this hook as well.
By using the existing scrolling event, you allow to decouple the virtualisation of this feature. So if I refactor the virtualisation and render just one div with absolute position for example, then I will just need to keep firing the event with the right contract to make sure this feature still works.

Reply:
Should we rename the events to avoid potential confusion?

Reply:
```
GRID_SCROLL -> GRID_NATIVE_SCROLL
GRID_SCROLLING -> GRID_ROWS_SCROLL //avoid all confusion, not the headers... the rows!
```

Reply:
Should I go ahead and rename those or open a separate PR for it? I agree with the new names.

Reply:
same PR is fine as it's a very close concern

Reply:
not resolved

Reply:
We don't need the change to ship value to the developers with this new callback. I would personally encourage doing it afterward **if** the diff is significant (e.g. +/-100 LOCs would likely signal that it's too large for a side improvement).

Reply:
Ok, @dtassone how about I add a third prop here https://github.com/mui-org/material-ui-x/blob/master/packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualRows.ts#L145 that is `absoluteTop` and just pass in the `scrollTop` position that has the correct value because changing the `top` breaks calculation for the scrollarea. That would allow me to use `useGridApiEventHandler(apiRef, GRID_ROWS_SCROLL, handler);` in the Infinite Loader hook?

Reply:
Updated